packages/web3/src/utils/sign.ts

```diff
@@ -36,11 +36,18 @@ necc.utils.hmacSha256Sync = (key: Uint8Array, ...messages: Uint8Array[]): Uint8A
   return Uint8Array.from(hash.digest())
 }
 
+function checkKeyType(keyType: KeyType) {
+  if (keyType !== 'default' && keyType !== 'bip340-schnorr' && keyType !== 'gl-secp256k1') {
```

Comment:
Is this condition still correct?

Reply:
Yes, because we currently only support the secp256k1 and schnorr, perhaps we can also add support for secp256r1 and ed25519 in the future.